src/coreclr/ToolBox/superpmi/superpmi-shim-collector/icorjitinfo.cpp

```diff
@@ -1203,12 +1203,31 @@ CorInfoTypeWithMod interceptor_ICJI::getArgType(CORINFO_SIG_INFO*       sig,
     },
     [&](DWORD exceptionCode)
     {
+
         this->mc->recGetArgType(sig, args, vcTypeRet, temp, exceptionCode);
     });
 
     return temp;
 }
 
+uint32_t interceptor_ICJI::getLoongArch64PassStructInRegisterFlags(CORINFO_CLASS_HANDLE cls)
+{
+
+    uint32_t temp = 0;
+    RunWithErrorExceptionCodeCaptureAndContinue(
+    [&]()
+    {
+        mc->cr->AddCall("getLoongArch64PassStructInRegisterFlags");
+        temp = original_ICorJitInfo->getLoongArch64PassStructInRegisterFlags(cls);
+    },
+    [&](DWORD exceptionCode)
+    {
+        this->mc->recGetFieldSizeClassificationByHnd(cls, temp);
+    });
+
+    return temp;
+}
+
```

Comment:
Move this after `getSystemVAmd64PassStructInRegisterDescriptor`

Also, this API will not ever throw an exception, I hope (and expect). Thus, you don't need all the exception code. Instead:

```suggestion
uint32_t interceptor_ICJI::getLoongArch64PassStructInRegisterFlags(CORINFO_CLASS_HANDLE cls)
{
    mc->cr->AddCall("getLoongArch64PassStructInRegisterFlags");
    uint32_t temp = original_ICorJitInfo->getLoongArch64PassStructInRegisterFlags(cls);
    mc->recGetLoongArch64PassStructInRegisterFlags(cls, temp);
    return temp;
}
```

Reply:
Thanks for your guidance.